drivers/adc/Kconfig

```diff
@@ -13,10 +13,16 @@ menuconfig ADC
 	bool
 	prompt "ADC drivers"
```

Comment:
add `select HAS_DTS_ADC` so that all ADC drivers require DTS support.

Reply:
But what would be the point in having `HAS_DTS_ADC` at all then? Right now it only excludes from Kconfig the options providing information that can be taken from DT, and these options are dependent on the same menuconfig entry that would select `HAS_DTS_ADC`. They should be rather removed then.
And this would be inconsistent with what we've got in other drivers. Is this a new approach that we want to take?

Reply:
As agreed on the API call, I'll add another commit with this change.